src/legacy/core_plugins/metrics/public/visualizations/components/flot_chart.js

```diff
@@ -92,14 +93,14 @@ class FlotChart extends Component {
 
   componentWillReceiveProps(newProps) {
     if (this.plot) {
-      const { series } = newProps;
+      const series = _.cloneDeep(newProps.series);
```

Comment:
You don't need to deepClone the whole `series` prop (we already have some performance problems because of too much cloning before mutate objects in kibana). You are just changing the fields `lines` and `bars`, 
so instead of  mutating the series on the `calculateData` function, you can just return on the `map` the following:
```js
if (_.isPlainObject(set)) {
   return {
      ...set,
      lines: computeColor(set.lines, set.color),
      bars: computeColor(set.bars, set.color),
    };
}
```
where `computeLinesColor` and `computeBarsColor` return an object like the original one:
```js
computeColor(style, color) {
  if (style && style.show) {
      const { fill, fillColor } = calculateFillColor(color, style.fill);
      return {
        ...style,
        fill,
        fillColor,
      };
  }
  return original;
}

Reply:
yep, looks ok for me, thanks. PR was updated